test/test_distributions.py

```diff
@@ -529,7 +529,7 @@ def test_beta_log_prob(self):
             x = dist.sample()
             actual_log_prob = dist.log_prob(x).sum()
             expected_log_prob = scipy.stats.beta.logpdf(x, alpha, beta)[0]
-            self.assertAlmostEqual(actual_log_prob, expected_log_prob, places=3)
+            self.assertAlmostEqual(actual_log_prob, expected_log_prob, places=3, allow_inf=True)
```

Comment:
I don't think we wanted to allow inf here by default, right @fritzo? We can have a test with values out of range, but in general always getting an inf makes the test pointless

Reply:
Hmm I think this is fine since we're comparing against a reference implementation: `scopy.stats.beta.logpdf()` should only return inf for extreme values of alpha and beta (e.g. alpha=1e5 and beta=1e-5) resulting in a very sharply peaked distribution.

Also this test should be less flaky after fixing random seeds in #4357. (sorry for creating a flaky test!)

Reply:
Yeah, I checked to see if `beta_log_prob` could return infinity before adding `allow_inf`, and it does seem to be permissible.